src/main/java/com/celements/navigation/TreeNode.java

```diff
@@ -156,19 +156,23 @@ public boolean equals(Object obj) {
     if (this == obj) {
       return true;
     }
-    if ((obj == null) || (obj.getClass() != this.getClass())) {
+    if (!(obj instanceof TreeNode)) { // null check included
       return false;
     }
     // object must be Test at this point
     TreeNode node = (TreeNode) obj;
-    return docRef.equals(node.docRef) && (position == node.position);
+    if (docRef.equals(node.docRef)) {
+      return (((position == null) && (node.position == null)) || ((position != null)
+          && position.equals(node.position)));
```

Comment:
There is a neat method `java.util.Objects.equals(a, b)` (since Java 7) which handles the null checks as well.